libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryUtils.java

```diff
@@ -177,6 +177,10 @@ static boolean isEmpty(String string) {
   }
 
   static String retrieveVendorId() {
+    if (MapboxTelemetry.applicationContext == null) {
+      return updateVendorId();
```

Comment:
Have we considered returning `obtainUniversalUniqueIdentifier()` here? Doing so, would remove the need for https://github.com/mapbox/mapbox-events-android/blob/329a1d657bbcf6ff555d64ef028d1a16f2c1d3ec/libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryUtils.java#L199-L203 in `updateVendorId()` 👇